/*
 * This file was automatically generated by EvoSuite
 * Mon Mar 05 22:44:26 GMT 2018
 */

package java_programs;

import org.junit.Test;
import static org.junit.Assert.*;
import java_programs.KNAPSACK;

public class KNAPSACK_ESTest {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int[][] intArray0 = new int[2][1];
      int[] intArray1 = new int[6];
      intArray1[0] = 1;
      intArray1[1] = 1;
      intArray1[2] = 1;
      intArray1[3] = 0;
      intArray1[4] = 1;
      intArray1[5] = 1;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[2];
      intArray2[0] = 1;
      intArray2[1] = 2466;
      intArray0[1] = intArray2;
      KNAPSACK.knapsack(1, intArray0);
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      KNAPSACK.knapsack(1200, intArray0);
      KNAPSACK.knapsack(1, intArray0);
      // Undeclared exception!
      KNAPSACK.knapsack(2466, intArray0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
      int[][] intArray0 = new int[4][4];
      int[] intArray1 = new int[4];
      intArray1[0] = 1906;
      intArray1[1] = 1906;
      intArray1[2] = 1906;
      intArray1[3] = 1906;
      intArray0[0] = intArray1;
      int[] intArray2 = new int[2];
      intArray2[0] = 1906;
      intArray2[1] = 1906;
      intArray0[1] = intArray2;
      int[] intArray3 = new int[8];
      intArray3[0] = 1906;
      intArray3[1] = 1906;
      intArray3[2] = 1906;
      intArray3[3] = 1906;
      intArray3[4] = (-469);
      intArray3[5] = 1906;
      intArray3[6] = 1906;
      intArray3[7] = 1906;
      intArray0[2] = intArray3;
      int[] intArray4 = new int[4];
      intArray4[0] = 1906;
      intArray4[1] = 1906;
      intArray4[2] = (-1);
      intArray4[3] = (-469);
      intArray0[3] = intArray4;
      int int0 = KNAPSACK.knapsack(1906, intArray0);
      assertEquals(1906, int0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int[][] intArray0 = new int[1][9];
      KNAPSACK.knapsack(1526, intArray0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(0, (int[][]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      int[][] intArray0 = new int[1][9];
      int[] intArray1 = new int[1];
      intArray1[0] = 1526;
      intArray0[0] = intArray1;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(1526, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1
         //
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      int[][] intArray0 = new int[5][5];
      int[] intArray1 = new int[0];
      intArray0[0] = intArray1;
      int[] intArray2 = new int[5];
      intArray2[0] = (-1443);
      intArray2[1] = (-1443);
      intArray2[2] = (-1443);
      intArray2[3] = (-1443);
      intArray2[4] = (-1443);
      intArray0[1] = intArray2;
      int[] intArray3 = new int[7];
      intArray3[0] = (-1443);
      intArray3[1] = (-1443);
      intArray3[2] = (-1443);
      intArray3[3] = (-1443);
      intArray3[4] = (-1443);
      intArray3[5] = (-1443);
      intArray3[6] = (-1443);
      intArray0[2] = intArray3;
      int[] intArray4 = new int[7];
      intArray4[0] = (-1443);
      intArray4[1] = (-1443);
      intArray4[2] = (-1443);
      intArray4[3] = (-1443);
      intArray4[4] = 205;
      intArray4[5] = (-1443);
      intArray4[6] = (-1443);
      intArray0[3] = intArray4;
      int[] intArray5 = new int[8];
      intArray5[0] = (-1443);
      intArray5[1] = 205;
      intArray5[2] = (-1746);
      intArray5[3] = 205;
      intArray5[4] = (-1443);
      intArray5[5] = 205;
      intArray5[6] = (-1443);
      intArray5[7] = 1;
      intArray0[4] = intArray5;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-1443), intArray0);
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }
}
